api/nodes/views.py

```diff
@@ -84,6 +88,35 @@ def get_node(self, check_object_permissions=True):
         return node
 
 
+class DraftMixin(object):
+
+    serializer_class = DraftRegistrationSerializer
+
+    def get_draft(self, draft_id=None):
+        node_id = self.kwargs['node_id']
+        if draft_id is None:
+            draft_id = self.kwargs['draft_id']
+        draft = get_object_or_error(DraftRegistration, draft_id)
+
+        if not draft.branched_from._id == node_id:
+            raise ValidationError('This draft registration is not created from the given node.')
+
+        if self.request.method not in drf_permissions.SAFE_METHODS:
+            registered_and_deleted = draft.registered_node and draft.registered_node.is_deleted
+
+            if draft.registered_node and not registered_and_deleted:
```

Comment:
Well I can't say `if not registered_and_deleted` because that would mean that would require that every draft has a node that has been registered and then deleted.  I want to check if there's a registered node in the first place.